vowpalwabbit/marginal.cc

```diff
@@ -221,9 +221,11 @@ void predict_or_learn(data& sm, LEARNER::base_learner& base, example& ec)
   
 void finish(data& sm)
 { sm.marginals.~unordered_map();
-  sm.expert_state.~unordered_map();
-  for (size_t i =0; i < 256; i++)
-    sm.temp[i].delete_v();
+  if (sm.compete) {
+    sm.expert_state.~unordered_map();
```

Comment:
I changed to 

void finish(data& sm)
{ sm.marginals.~unordered_map();
  if (sm.compete) 
    sm.expert_state.~unordered_map();
 for (size_t i =0; i < 256; i++)
    sm.temp[i].delete_v();
}